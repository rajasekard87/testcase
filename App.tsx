import React, { useState } from 'react';
import {
  View,
  Text,
  TextInput,
  Button,
  TouchableOpacity,
  StyleSheet,
} from 'react-native';

const App = (props: { login: any; }) => {

  const [username, _setUsername] = useState('');
  const [password, _setPassword] = useState('');
  const [phone, _setPhone] = useState('');

  const _onSubmitLogin = () => {
    const { login } = props;
    if (login) {
      login({ username, password });
    }
  }

  return (
    <View style={styles.container}>
<Text style={styles.title}> Welcome Back</Text>
<View style={styles.inputView}>
<TextInput
style={styles.inputText}
testID="username"
value={username}
placeholder="Enter Username"
placeholderTextColor="#003f5c"
onChangeText={text => _setUsername(text)}
/>
</View>
<View style={styles.inputView}>
<TextInput
testID="password"
value={password}
style={styles.inputText}
secureTextEntry
placeholder="Enter Password"
placeholderTextColor="#003f5c"
onChangeText={text => _setPassword(text)}/>
</View>
<TouchableOpacity
testID="btnSubmit"
onPress={_onSubmitLogin}
style={styles.loginBtn}>
<Text style={{color:"white",fontSize:18}}>LOGIN</Text>
</TouchableOpacity>
<TouchableOpacity
onPress = {()=>{}}>
<Text style={styles.forgotAndSignUpText}>New Here ? Register</Text>
</TouchableOpacity>
</View>
  );
}

const styles = StyleSheet.create({
  container: {
  flex: 1,
  backgroundColor: '#ffffff',
  alignItems: 'center',
  justifyContent: 'center',
  },
  title:{
  fontWeight: "bold",
  fontSize:30,
  color:"#20b2aa",
  marginBottom: 40, 
  marginTop:-20
  },
  inputView:{
  width:"80%",
  backgroundColor:"lightgrey",
  borderRadius:25,
  height:50,
  marginBottom:20,
  justifyContent:"center",
  padding:20
  },
  inputText:{
  height:50,
  color:"black"
  },
  forgotAndSignUpText:{
  color:"#20b2aa",
  fontSize:13,
  margin:10,
  fontWeight:'bold'
  },
  loginBtn:{
  width:"80%",
  backgroundColor:"#20b2aa",
  borderRadius:25,
  height:50,
  alignItems:"center",
  justifyContent:"center",
  marginTop:20,
  marginBottom:10
  },
  forgotText:{
  color:"black",
  fontSize:12,
  textAlign:"right",
  justifyContent:"flex-end",
  flexDirection:"row",
  alignItems:"flex-end"
  }
  });

export default App;

