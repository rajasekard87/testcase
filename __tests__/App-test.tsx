/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';
import { render, screen, fireEvent } from '@testing-library/react-native'

// it('renders correctly', () => {
  
//   const tree = render(<App login={undefined}/>);
//   expect(tree.toJSON()).toMatchSnapshot();

// });

describe('Login', () => {

  describe('change text login', () => {
    it('change text username and password', () => {
      const { getByTestId } = render(<App login={undefined} />);

      // use fireEvent change value TextInput
      fireEvent.changeText(getByTestId('username'), 'admin');
      fireEvent.changeText(getByTestId('password'), 'admin@123');

      // use toEqual check value TextInput
      expect(getByTestId('username').props.value).toEqual('admin');
      expect(getByTestId('password').props.value).toEqual('admin@123');
    });

  });
  describe('Submit form login', () => {

    it('on submit login', () => {
      const data = { "password": "123456", "username": "admin@123" }
      const submitHandler = jest.fn();
      const { getByTestId } = render(
        // passing prop to Login component
        <App login={submitHandler} />

      );
      fireEvent.changeText(getByTestId('username'), 'admin@123');
      fireEvent.changeText(getByTestId('password'), '123456');

      expect(getByTestId('username').props.value).toEqual('admin@123');
      expect(getByTestId('password').props.value).toEqual('123456');

      // use fireEvent.press call Button submit
      fireEvent.press(getByTestId('btnSubmit'));

      // checking ouput data equal input
      expect(submitHandler).toHaveBeenCalledWith(data);
    });

  })

});