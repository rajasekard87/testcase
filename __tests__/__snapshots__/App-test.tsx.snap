// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "color": "#20b2aa",
        "fontSize": 30,
        "fontWeight": "bold",
        "marginBottom": 40,
        "marginTop": -20,
      }
    }
  >
     Welcome Back
  </Text>
  <View
    style={
      {
        "backgroundColor": "lightgrey",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "marginBottom": 20,
        "padding": 20,
        "width": "80%",
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Enter Username"
      placeholderTextColor="#003f5c"
      style={
        {
          "color": "black",
          "height": 50,
        }
      }
      testID="username"
      value=""
    />
  </View>
  <View
    style={
      {
        "backgroundColor": "lightgrey",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "marginBottom": 20,
        "padding": 20,
        "width": "80%",
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Enter Password"
      placeholderTextColor="#003f5c"
      secureTextEntry={true}
      style={
        {
          "color": "black",
          "height": 50,
        }
      }
      testID="password"
      value=""
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#20b2aa",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginTop": 20,
        "opacity": 1,
        "width": "80%",
      }
    }
    testID="btnSubmit"
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 18,
        }
      }
    >
      LOGIN
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        {
          "color": "#20b2aa",
          "fontSize": 13,
          "fontWeight": "bold",
          "margin": 10,
        }
      }
    >
      New Here ? Register
    </Text>
  </View>
</View>
`;
